### PR TITLE
Dynamics

### PR DESCRIPTION
**Pull request by bwrsandman**
Adding physics with Bullet.

I'm testing with bullet 3 but it should be compatible with bullet 2.

So far I have:
- [x] Land as static collision mesh
- [x] Ray casting the mouse to the land
- [x] Using surface normal to orient the debug cross

### CODE REVIEW

**Comment by bwrsandman** [comment]
You can now gather information about objects by raycasting them.
I don't know entt enough to get more information about entities so that's left as an assignment.

Screenshot from 2020-05-20 17-50-34

**Comment by bwrsandman** [comment]
Rebased to and tested current master.
Failing vcpkg linking on linux seems to be a vcpkg package bug. When linking system libs it works. 
Vcpkg issue happens when I test it too on latest vcpkg master.

**Review by raffclar — COMMENTED** [review]
Compiled, ran, tested a few islands with no issues. Is it worth including info in the profiler?

File: src/Dynamics/LandBlockBulletMeshInterface.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			_indices[i] = i;
+		}
+	}
+
+	/// get read and write access to a subpart of a triangle mesh
+	/// this subpart has a continuous array of vertices and indices
+	/// in this way the mesh can be handled as chunks of memory with striding
+	/// very similar to OpenGL vertexarray support
+	/// make a call to unLockVertexBase when the read and write access is finished
+	void getLockedVertexIndexBase(unsigned char** vertexbase, int& numverts, PHY_ScalarType& type, int& stride,
```
Comment (raffclar) [inline src/Dynamics/LandBlockBulletMeshInterface.h:47]: Lots of warnings about unused parameters

**Comment by bwrsandman** [comment]
Definitely should include profiler

**Review by raffclar — APPROVED** [review]

**Pull request merged**